test/test_jit.py

```diff
@@ -8288,17 +8288,36 @@ def as_interface(x):
                    .check("Traceback") \
                    .check("in foo").check("in baz").run(str(cm.exception))
 
-    def test_binop_unsupported_error(self):
-        with self.assertRaisesRegex(NotSupportedError, "unsupported binary operator:"):
-            @torch.jit.script
-            def binop(x, y):
-                # Replace this with another unsupported op when/if it gets supported
-                return x << y
+    def test_operator_precedence(self):
+        def double(x):
+            return 2 * x
+
+        def complicated_arithmetic_operation():
+            list = [0, 1, 2, 3]
+            result = list[1:3][0] + 3 ** 2 - double(4) + (-3 + 4) * 6 // 2 % 2 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + ~4
+            # result = list[1:3][0] + 3 ** 2 - double(4) + 1 * 6 // 2 % 2 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + ~4
+            # result = list[1:3][0] + 3 ** 2 - 8 + 1 * 6 // 2 % 2 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + ~4
+            # result = [1, 2][0] + 3 ** 2 - 8 + 1 * 6 // 2 % 2 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + ~4
+            # result = 1 + 3 ** 2 - 8 + 1 * 6 // 2 % 2 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + ~4
+            # result = 1 + 9 - 8 + 1 * 6 // 2 % 2 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + ~4
+            # result = 1 + 9 - 8 + 1 * 6 // 2 % 2 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + -5
+            # result = 1 + 9 - 8 + 6 // 2 % 2 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + -5
+            # result = 1 + 9 - 8 + 3 % 2 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + -5
+            # result = 1 + 9 - 8 + 1 << 2 + 1 >> 1 & 4 ^ 2 + 1 | 13 + -5
+            # result = 3 << 3 >> 1 & 4 ^ 3 | 8
+            # result = 24 >> 1 & 4 ^ 3 | 8
+            # result = 12 & 4 ^ 3 | 8
+            # result = 4 ^ 3 | 8
+            # result = 7 | 8
+            # result = 15
+            return result
+
+        self.assertEqual(complicated_arithmetic_operation(), 15)
```

Comment:
`self.checkScript(some_python_function, [input1, input2])` will automatically compile the function and check the TorchScript outputs against Python's, can you use that here instead?